### ISSUE TITLE
Modify module variables from Python using setters

### ISSUE THREAD

**Issue by yguclu** [pyccel/pyccel#742]
As a USER, I would like that my Python code could modify the module variables of a C Python extension module generated by Pyccel.

**Comment by EmilyBourne** [comment]
Blocked by #886

**Comment by EmilyBourne** [comment]
Doc can be found [here](https://docs.python.org/3/c-api/typeobj.html#c.setattrfunc) and [here](https://docs.python.org/3/extending/newtypes.html#attribute-management)

**Comment by EmilyBourne** [comment]
This does not seem possible. Python exposes module variables in the same way it exposes all variables. That is to say that it creates a label that can be pointed to any object. Variables can be added to a module using the [PyModule_AddObject](https://docs.python.org/3/c-api/module.html#c.PyModule_AddObject) method, however these variables are normal python objects which cannot modify an underlying variable of a basic type (it should still be possible to modify elements of an array).

This can be seen for example in the [math](https://github.com/python/cpython/blob/main/Modules/mathmodule.c#L3520) module. The math module exposes the `pi` variable as a `PyFloat` and uses it internally in functions such as the `sin` function
```
In [1]: import math                                                   

In [2]: math.pi                                                       
Out[2]: 3.141592653589793

In [3]: math.pi = True                                                

# math.pi value can be changed to anything
In [4]: math.pi                                                       
Out[4]: True

# sin function is unaffected 
In [5]: math.sin(3.141592653589793238462643383279502884197)          
Out[5]: 1.2246467991473532e-16
```

I cannot think of a way to work around this restriction without preventing the function from being called from another module

**Comment by yguclu** [comment]
Here is an example for a directory that contains two files.

File **a.py**:
```python
pi = 3.14
```

File **b.pi**:
```python
from a import pi
```

Test with `ipython`:
```ipython
In [1]: import a                                                                

In [2]: a.pi                                                                    
Out[2]: 3.14

In [3]: a.pi = True                                                             

In [4]: import b                                                                

In [5]: b.pi                                                                    
Out[5]: True
```

**Comment by EmilyBourne** [comment]
> Within the module scope, `math.pi` has the original float value. But within the global scope, any other module that imports `math` will see that `math.pi = True`.

Exactly, and this is the behaviour that we cannot replicate. It may be replicated on the python side, however if we translate a file which uses the constant then it will use the definition from the module scope not the global scope
